web-common/src/features/dashboards/dimension-table/export-toplist.ts

```diff
@@ -0,0 +1,63 @@
+import type { TimeControlStore } from "@rilldata/web-common/features/dashboards/time-controls/time-control-store";
+import type {
+  V1ExportFormat,
+  createQueryServiceExport,
+} from "@rilldata/web-common/runtime-client";
+import { get } from "svelte/store";
+import { runtime } from "../../../runtime-client/runtime-store";
+import { useDashboardStore } from "../dashboard-stores";
+import { getQuerySortType } from "../leaderboard/leaderboard-utils";
+import { SortDirection } from "../proto-state/derived-types";
+
+export default async function exportToplist({
+  query,
+  metricViewName,
+  format,
+  timeControlStore,
+}: {
+  query: ReturnType<typeof createQueryServiceExport>;
+  metricViewName: string;
+  format: V1ExportFormat;
+  // we need this from argument since getContext is called to get the state managers
+  // which cannot run outside of component initialisation
+  timeControlStore: TimeControlStore;
```

Comment:
Can we call `getStateManagers()` inside the Export button component and pass that to the query here so that we have `ctx` for all the required parameters apart from `format`?

Reply:
Good callout. Admittedly I just replicated the approach taken in `rows-viewer/export-metrics.ts`.